src/Microsoft.Diagnostics.Runtime.Tests/src/MemoryReaderTests.cs

```diff
@@ -43,11 +43,11 @@ public void SearchMemoryTest()
                         if (!dt.DataReader.Read(expectedOffset, slice, out int read) || read != slice.Length)
                             continue;
 
-                        ulong addressFound = dt.DataReader.SearchMemory(segment.Start, segment.End, buffer.Slice(0, i + sizeof(ulong)));
+                        ulong addressFound = dt.DataReader.SearchMemory(segment.Start, (int)segment.Length, buffer.Slice(0, i + sizeof(ulong)));
 
                         // There could still accidently be a pattern that matches this somewhere
                         while (addressFound != 0 && addressFound < expectedOffset)
-                            addressFound = dt.DataReader.SearchMemory(addressFound + 1, segment.End, buffer.Slice(0, i + sizeof(ulong)));
+                            addressFound = dt.DataReader.SearchMemory(addressFound + 1, (int)segment.End, buffer.Slice(0, i + sizeof(ulong)));
```

Comment:
Arithmetic is needed here.

Well, from the uses of that method it does seem that accepting an end address is more natural... I have no idea why most APIs in the core library prefer a length though.

Reply:
It's surprising that the test passed.  I must have gotten lucky with that underflow.

Reply:
Taking a length makes a lot of sense when working with strings, arrays, and lists.  I'm happy to follow that convention here.  I just need more coffee to stop failing at basic coding.  =)